File: backend/app/api/five_seconds.py
from fastapi import APIRouter, HTTPException

from backend.app.core.five_seconds_prompt_loader import (
    FiveSecondsPromptLoaderError,
    load_five_seconds_prompts,
)
from backend.app.models.five_seconds import FiveSecondsPrompt

router = APIRouter(prefix="/api/5-seconds", tags=["5-seconds"])


@router.get("/prompts", response_model=list[FiveSecondsPrompt])
def list_five_seconds_prompts() -> list[FiveSecondsPrompt]:
    """Return the authored prompt bank for the 5 Seconds game."""

    try:
        return load_five_seconds_prompts()
    except FiveSecondsPromptLoaderError as error:
        raise HTTPException(status_code=500, detail=str(error)) from error

File: frontend/src/api/fiveSeconds.ts
import { API_BASE_URL } from './config'
import type { FiveSecondsPrompt } from '../types/fiveSeconds'

/** Fetch the authored prompt bank for the 5 Seconds game. */
export async function fetchFiveSecondsPrompts() {
  const response = await fetch(`${API_BASE_URL}/api/5-seconds/prompts`)
  if (!response.ok) {
    throw new Error('Nie udało się pobrać promptów gry 5 sekund.')
  }

  return (await response.json()) as FiveSecondsPrompt[]
}
